crates/notion-core/src/manifest/serial.rs

```diff
@@ -93,6 +94,8 @@ impl Manifest {
             return Ok(Some(image::Image {
                 node: VersionSpec::parse_version(&toolchain.node)?,
                 node_str: toolchain.node.clone(),
+                npm: VersionSpec::parse_version(&toolchain.npm)?,
```

Comment:
I'm still coming up to speed on both Rust and the structure of Notion, but does this mean that if `npm` is not present in `toolchain` that loading the manifest will return an error? Which will mean that none of the tools will be able to execute because they won't see an active image for the current directory?

I see there's code later on that when you pin the version of Node, it will also write the NPM version, but if I'm correct in my interpretation above, this means that devs won't be able to just manually edit `package.json` to add `toolchain: { node: "10.0.0" }` or whatever, without also including the exact version of NPM that is bundled with that version of Node. Instead, the only simple way to enable Notion in a project is with the `pin` command.

From an ergonomics standpoint, it seems like it would be nice to continue to allow a minimal `node` configuration so that Notion can be enabled manually instead of essentially requiring the user to run a command. I could also see confusion for a user that is using `yarn`, asking "Why is there an NPM configuration added to my project when I don't use NPM for anything?"

Reply:
@charlespierce Great question! There's a slightly uncomfortable mismatch here, in that the toolchain key is meant to fully pin all the information, so it's more like a lockfile, and yet it's stored in `package.json`, which is a human-owned file.

We could fudge a little bit and say that an absent `"npm"` key is either sugar for the default npm version that comes with the node version, or it means `"npm"` is not supported at all. Pro: lighter-weight syntax. Con: more variation means more ways to say the same thing, which also means automated commands for doing the same thing might either create churn in git commits or have to get smarter to preserve style.

But one decision we made earlier that I still think was the right one was disallowing loose semver specifications for the versions -- you can only fully specify an exact pinned version; semver resolution is never triggered by reading the manifest. This is the right call because it's the basis for reproducibility.

It's worth at least asking whether this all calls for a separate notion lockfile. I'm just very hesitant to impose another file on JS projects for such a small amount of data.

So maybe a good middle ground to start with for now (which we can revisit after usage experience) is:

- `"npm"` defaults to the default version that comes with Node, because that's the most common usage pattern
- `"yarn"` defaults to not-present, since it's not Node's default package manager
- when Notion writes to `package.json`, it should stylistically prefer omitting the `"npm"` key if it's the same as the default version, to create a pleasant canonical style and minimize churn in git commits
 
Thoughts?

Reply:
Those seem like good defaults that maintain the simple path for devs while keeping the necessary specificity that we want inside of Notion.

One thought I had from looking at the way the `notion_home` folder is structured with this PR, is that an unspecified `npm` version in `package.json` could be mapped to a folder named e.g. `default`. That way internally the representation can still be that there is no specific `npm` version, but it will correctly map to a filesystem location.